src/log.ts

```diff
@@ -0,0 +1,189 @@
+function log (str: string, type: number = TMESG, typestr: string = 'mesg') {
+  if (!(_level & type)) return // bitmask
+
+  if (typeof _dom !== 'undefined') {
+    const logArea = _dom.getElementById('status')
+    if (!logArea) return
+
+    const addendum = _dom.createElement('span')
+    addendum.setAttribute('class', typestr)
+    const now = new Date()
+    addendum.innerHTML = `${now.getHours()}:${now.getMinutes()}:${now.getSeconds()} [${typestr}] ${escapeForXML(str)}<br/>`
+    if (_ascending) {
+      logArea.insertBefore(addendum, logArea.firstChild)
+    } else {
+      logArea.appendChild(addendum)
+    }
+  } else if (typeof console !== 'undefined') {
+    console.log(str)
+  }
+}
+
+/**
+ * Adds a message to the element with id "status". The messages are prepended with
+ * time and type of message, in this case [mesg].
+ */
+export function msg (message: string) {
+  log(message)
+}
+
+/**
+ * Adds a warning message to the element with id "status". The messages are
+ * prepended with time and type of message, in this case [warn].
+ */
+export function warn (message: string): void {
+  log(message, TWARN, 'warn')
+}
+
+/**
+ * Adds a debugging message to the element with id "status". The messages are
+ * prepended with time and type of message, in this case [dbug].
+ */
+export function debug (message: string): void {
+  log(message, TDEBUG, 'dbug')
+}
+
+/**
+ * Adds a info message to the element with id "status". The messages are
+ * prepended with time and type of message, in this case [info].
+ */
+export function info (message: string): void {
+  log(message, TINFO, 'info')
+}
+
+/**
+ * Adds a error to the element with id "status". The messages are
+ * prepended with time and type of message, in this case [eror].
+ */
+export function error (message: string): void {
+  log(message, TERROR, 'eror')
+}
+
+/**
+ * Adds a success message to the element with id "status". The messages are
+ * prepended with time and type of message, in this case [good].
+ */
+export function success (message: string): void {
+  log(message, TSUCCESS, 'good')
+}
+
+/**
+ * Uses the global alert to send an alert. If global alert is not available, it
+ * will output the message using the method [[warning]]s.
+ */
+export function alert (message: string): void {
+  if (_window && typeof _window.alert !== 'undefined') {
+    _window.alert(message)
+  } else {
+    warn(message)
+  }
+}
+
+/**
+ * Will clear the content of the element with id "status".
+ */
+export function clear (): void {
+  const logArea = _dom?.getElementById('status')
+  if (!logArea) return
+  logArea.innerHTML = ''
+}
+
+/**
+ * Lets you configure which types of messages will be shown. The number is a
+ * sum of the levels you want to show:
+ *
+ * - Error: 1
+ * - Warning: 2
+ * - Message: 4
+ * - Success: 8
+ * - Info: 16
+ * - Debug: 32
+ *
+ * You can also choose to set all by passing the number *64*.
+ */
+export function setLevel (level: number): void {
+  _level = TALL
```

Comment:
What does this line do?

Reply:
Just making sure that the next line are shown, could be set to TDEBUG instead.